test/mechanics/integrands/HistoryData.cpp

```diff
@@ -0,0 +1,397 @@
+        momentumBalanceCells.Add(cellContainer.back());
+    }
+
+    // Initialize IP data %%%%%%%%%%%%%%%%%%%%%%%
+    creepLaw.InitializeHistoryData(momentumBalanceCells, displ);
+
+
+    // Get gradient %%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    SimpleAssembler assembler(dofInfo.numIndependentDofs, dofInfo.numDependentDofs);
+    VectorXd displacements = VectorXd::Zero(dofInfo.numIndependentDofs[displ]);
+
+    // Build external Force %%%%%%%%%%%%%%%%%%%%%
+    constexpr double rhsForce = 2000.;
+    GlobalDofVector extF;
+    extF.J[displ].setZero(dofInfo.numIndependentDofs[displ]);
+    extF.K[displ].setZero(dofInfo.numDependentDofs[displ]);
+    NodeSimple& nodeRight = mesh.NodeAtCoordinate(Eigen::VectorXd::Ones(1) * SpecimenLength, displ);
+    extF.J[displ][nodeRight.GetDofNumber(0)] = rhsForce;
+
+    // Post processing stuff %%%%%%%%%%%%%%%%%%%%
+    std::vector<double> rhsDispNumerical;
+
+    // Time loop %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    constexpr double timeFinal = 10.;
+    constexpr unsigned int maxIter = 10;
+    double time = 0.;
+    delta_t = 0.01;
+
+    while (time < timeFinal)
+    {
+        unsigned int numIter = 0;
+        time += delta_t;
+
+        // Calculate residual %%%%%%%%%%%%%%%%%%%
+        GlobalDofVector gradient = assembler.BuildVector(momentumBalanceCells, {displ}, MomentumGradientF);
+        Eigen::VectorXd residual = gradient.J[displ] + extF.J[displ];
+
+        // Iterate for equilibrium %%%%%%%%%%%%%%
+        while (numIter < maxIter && residual.lpNorm<Infinity>() > 1e-9)
+        {
+            numIter++;
+            // Build and solve system %%%%%%%%%%%
+            GlobalDofMatrixSparse hessian = assembler.BuildMatrix(momentumBalanceCells, {displ}, MomentumHessian0F);
+            Eigen::MatrixXd hessianDense(hessian.JJ(displ, displ));
+            Eigen::VectorXd deltaDisplacements = hessianDense.ldlt().solve(residual);
+            displacements -= deltaDisplacements;
+
+            // Merge dof values %%%%%%%%%%%%%%%%%
+            int numUnconstrainedDofs = dofInfo.numIndependentDofs[displ];
+            for (NodeSimple& node : mesh.NodesTotal(displ))
+            {
+                int dofNumber = node.GetDofNumber(0);
+                if (dofNumber < numUnconstrainedDofs)
+                    node.SetValue(0, displacements[dofNumber]);
+            }
+
+            // Calculate new residual %%%%%%%%%%%
+            gradient = assembler.BuildVector(momentumBalanceCells, {displ}, MomentumGradientF);
+            residual = gradient.J[displ] + extF.J[displ];
+        }
+        if (numIter >= maxIter)
+        {
+            std::cout << residual.lpNorm<Infinity>() << std::endl;
+            std::cout << time << std::endl;
+            throw Exception(__PRETTY_FUNCTION__, "No convergence");
+        }
+        // Update history data %%%%%%%%%%%%%%%%%%
+        for (auto& cell : momentumBalanceCells)
+            cell.Apply(MomentumUpdateHistoryDataF);
```

Comment:
Similar here, delta_t should be an input to the function.

Reply:
See above. Our generalized cell is not interested in a delta_t. We can change that if that's what everybody wants, but this topic is not related to history data.